azure/scope/machine.go

```diff
@@ -269,9 +269,6 @@ func (m *MachineScope) AvailabilityZone() string {
 	if m.AzureMachine.Spec.FailureDomain != nil {
 		return *m.AzureMachine.Spec.FailureDomain
 	}
```

Comment:
small nit

remove the comment on this func about `AzureMachine.Spec.AvailabilityZone.ID (This is DEPRECATED)`